src/main/kotlin/com/hypto/iam/server/service/UsersService.kt

```diff
@@ -2,41 +2,61 @@
 package com.hypto.iam.server.service
 
 import com.google.gson.Gson
+import com.hypto.iam.server.configs.AppConfig
 import com.hypto.iam.server.db.repositories.OrganizationRepo
+import com.hypto.iam.server.db.repositories.UserRepo
+import com.hypto.iam.server.db.tables.pojos.Users
 import com.hypto.iam.server.exceptions.EntityNotFoundException
 import com.hypto.iam.server.extensions.toUserStatus
 import com.hypto.iam.server.idp.IdentityGroup
 import com.hypto.iam.server.idp.IdentityProvider
 import com.hypto.iam.server.idp.NextToken
 import com.hypto.iam.server.idp.PasswordCredentials
 import com.hypto.iam.server.idp.RequestContext
+import com.hypto.iam.server.idp.UserAlreadyExistException
 import com.hypto.iam.server.models.BaseSuccessResponse
 import com.hypto.iam.server.models.PaginationOptions
 import com.hypto.iam.server.models.UpdateUserRequest
 import com.hypto.iam.server.models.User
 import com.hypto.iam.server.utils.IamResources
 import com.hypto.iam.server.utils.ResourceHrn
+import java.time.LocalDateTime
 import org.koin.core.component.KoinComponent
 import org.koin.core.component.inject
 
 class UsersServiceImpl : KoinComponent, UsersService {
+    private val userRepo: UserRepo by inject()
     private val organizationRepo: OrganizationRepo by inject()
     private val identityProvider: IdentityProvider by inject()
     private val gson: Gson by inject()
+    private val appConfig: AppConfig by inject()
 
     override suspend fun createUser(
         organizationId: String,
         credentials: PasswordCredentials,
-        createdBy: String?
+        createdBy: String?,
+        verified: Boolean
     ): User {
         val org = organizationRepo.findById(organizationId)
             ?: throw EntityNotFoundException("Invalid organization id name. Unable to create a user")
+
+        if ((appConfig.app.uniqueUsersAcrossOrganizations && UserRepo.existsByEmail(credentials.email)) ||
```

Comment:
We will be un-necessarily querying db twice if the first condition fails. Rather the condition can be

```
if (appConfig.app.uniqueUsersAcrossOrganization)
   if (UserRepo.existByEmail(credentials.email)
      throw ex
} else {
    if (UserRepo.existsByEmail(email, orgId))
       throw ex
}
```